### PR TITLE
Makefile

### PR DESCRIPTION
**Pull request by KerimCETINBAS**
- [makefile.example](https://github.com/edunota/edunota-database-migrations/blob/makefile/makefile.example) created
- readme adjusted for makefile commands

create your own makefile based on [makefile.example](https://github.com/edunota/edunota-database-migrations/blob/makefile/makefile.example)  and edit the variables to match your env
```bash
POSTGRES_USER=edunota
# pasword of database
POSTGRES_PASSWORD=edunota
# database to apply migrations
POSTGRES_DB=edunota
# port of database
POSTGRES_PORT=5432
# hostname of database
POSTGRES_HOST=localhost
```

makefile.dev
```bash
make -f makefile.dev  create name=create_database
```

will add a new migration under /migrations dir something name match with 20230915015642_create_database

note: -f flag is optional if you present makefile in your root

makefile
```bash
make create name=create_database
````

### CODE REVIEW

**Comment by caglarcercinli** [comment]
i think we need a live demonstration for this. i dont get how it works like this.

**Comment by KerimCETINBAS** [comment]
> i think we need a live demonstration for this. i dont get how it works like this.

ofcourse

**Comment by KerimCETINBAS** [comment]
I created a pr creating very first tables to prevent conflict, marge #5 into [makefile branch](https://github.com/edunota/edunota-database/tree/makefile) before mergin this into main 
fyi

**Comment by caglarcercinli** [comment]
@KerimCETINBAS could you please update readme.md about ssh cloning?

**Comment by KerimCETINBAS** [comment]
@caglarcercinli 
> @KerimCETINBAS could you please update readme.md about ssh cloning?


i pushed a commit that update readme.md about ssh on #5 

added ssh guide on https://github.com/edunota/edunota-guideline/blob/main/ssh.md

File: makefile.example [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# used by goose, do not edit these fields																								    
+export GOOSE_DBSTRING=user=$(POSTGRES_USER) password=$(POSTGRES_PASSWORD) host=$(POSTGRES_HOST) port=$(POSTGRES_PORT) dbname=$(POSTGRES_DB)
+export GOOSE_DRIVER=postgres
+# ============================================
+
+# ==========================================================
+# commands
+# spin up database and other services on docker compose
+spinup: 
+	bash -c "sudo docker compose --project-directory=infrastructure/$(ENV) up -d"
```
Comment (FatiUnal) [inline makefile.example:37]: I am windows user . sudo is not a windows comment

**Review by FatiUnal — CHANGES_REQUESTED** [review]
` bash -c "sudo docker compose --project-directory=infrastructure/$(ENV) up -d" `
I am windows user . sudo is not windows command

File: makefile.example [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+# port of database
+POSTGRES_PORT=5432
+
+# hostname of database
+POSTGRES_HOST=localhost
+
+
+# ============================================
+# used by goose, do not edit these fields																								    
+export GOOSE_DBSTRING=user=$(POSTGRES_USER) password=$(POSTGRES_PASSWORD) host=$(POSTGRES_HOST) port=$(POSTGRES_PORT) dbname=$(POSTGRES_DB)
```
Comment (FatiUnal) [inline makefile.example:29]: 2023/09/17 22:12:35 goose run: failed to ensure DB version: failed to connect to `host=localhost user=edunota database=edunota`: failed SASL auth (FATAL: password authentication failed for user "edunota" (SQLSTATE 28P01))

File: makefile.example (hunk 1 above)
Comment (KerimCETINBAS) [reply makefile.example:37]: This is a problem caused by windows, better switch a unix environment. or create your own makefile based on example

**Review by caglarcercinli — APPROVED** [review]

**Pull request merged**